presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -326,7 +326,9 @@ public FSDataOutputStream create(Path path, FsPermission permission, boolean ove
             throw new IOException("File already exists:" + path);
         }
 
-        createDirectories(stagingDirectory.toPath());
+        if (!stagingDirectory.exists()) {
+            createDirectories(stagingDirectory.toPath());
+        }
```

Comment:
Add a check afterwards, in case it exists but is not a directory (or symlink to a directory):
```java
if (!stagingDirectory.isDirectory()) {
    throw new IOException("Staging directory is not a directory: " + stagingDirectory);
}
```
I suspect this works, but the test will confirm it.